UnityProject/Assets/Scripts/Items/Implants/Organs/Liver.cs

```diff
@@ -85,16 +85,16 @@ private void BloodToLiver(ReagentContainerBody blood)
 			//figure out how much we are going to process or remove
 			lock (blood.CurrentReagentMix.reagents)
 			{
-				foreach (Reagent reagent in blood.CurrentReagentMix.reagents.Keys)
+				foreach (var reagent in blood.CurrentReagentMix.reagents.m_dict)
```

Comment:
what's the ``m_dict``?

Reply:
it's the dictionary at the base of the serialisable dictionary

Reply:
serializable dictionaries are really bad with performance and gc, accessing it directly makes it way better in that regard, (m_dict was private previously)
I had a plan to remove them eventually